app/api/v2/work/route.js

```diff
@@ -0,0 +1,123 @@
+
+import { NextResponse } from "next/server";
+import prisma from "@/prisma/prismaClient";
+import { Auth } from "../../utils/auth";
+import { getServerSession } from "next-auth";
+import { authOptions } from "@/app/api/utils/authOptions";
+
+
+export async function POST(req) {
+
+  const session = await getServerSession(authOptions)
+  const params = await req.json()
+  const authCheck = new Auth(session, params)
+    .requireRoles([])
+    .requireParams(["loggedBy", "loggedFor", "workedAt", "duration", "description", "semesterId"])
+
+  if (authCheck.failed) return authCheck.verify(authCheck.response)
+
+
+
+  const success = await prisma.$transaction(async (transaction) => {
+    const workLogEntry = await transaction.workLog.create({
+      data: {
+        loggedBy: params.loggedBy,
+        loggedFor: params.loggedFor,
+        workedAt: params.workedAt,
+        duration: params.duration,
+        description: params.description,
+        semesterId: params.semesterId
+      }
+    });
+
+    const voucherBuffer = (await transaction.VoucherBuffer.findFirst({
+      select: {
+        buffer: true
+      },
+      where: {
+        userId: params.loggedFor
+      }
+    })).buffer;
+
+    const numVouchersToLog = (params.duration / 2) + voucherBuffer; // TODO: Update to make vouchers per hour ajustable and not hardcoded
+    
+    const loggedDate = new Date();
+    const workLogId = workLogEntry.id;
+
+    const voucherExpirationDate = (await transaction.Semester.findFirst({
+      select: {
+        voucherExpirationDate: true
+      },
+      orderBy:{
+        id: 'desc'
+      }
+    })).voucherExpirationDate
+
+    const voucherData = [];
+    for (let i = 0; i < numVouchersToLog; i++) {
```

Comment:
Sørg for at den ikke lager ekstra voucher når `numVoucherToLog` har desimal
```suggestion
    for (let i = 0; i < Math.floor(numVouchersToLog); i++) {
```